browser/ui/views/brave_vpn/brave_vpn_fallback_dialog_view.cc

```diff
@@ -0,0 +1,163 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/brave_vpn/brave_vpn_fallback_dialog_view.h"
+
+#include <memory>
+#include <utility>
+
+#include "brave/browser/brave_features_internal_names.h"
+#include "brave/components/brave_vpn/common/pref_names.h"
+#include "chrome/browser/about_flags.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/lifetime/application_lifetime.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "components/constrained_window/constrained_window_views.h"
+#include "components/flags_ui/pref_service_flags_storage.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/controls/button/checkbox.h"
+#include "ui/views/controls/link.h"
+#include "ui/views/controls/styled_label.h"
+#include "ui/views/layout/box_layout.h"
+
+namespace brave {
+void ShowBraveVpnIKEv2FallbackDialog() {
+  brave_vpn::BraveVpnFallbackDialogView::Show(chrome::FindLastActive());
+}
+}  // namespace brave
+
+namespace brave_vpn {
+
+namespace {
+
+constexpr char kBraveVPNLearnMoreURL[] =
+    "https://support.brave.com/hc/en-us/articles/";
+
+constexpr int kChildSpacing = 16;
+constexpr int kPadding = 24;
+constexpr int kTopPadding = 32;
+constexpr int kBottomPadding = 26;
+constexpr int kDialogWidth = 600;
+
+}  // namespace
+
+// static
+void BraveVpnFallbackDialogView::Show(Browser* browser) {
+  auto* prefs = browser->profile()->GetPrefs();
+  if (!prefs->GetBoolean(prefs::kBraveVPNWireguardFallbackDialog)) {
+    return;
+  }
+
+  // The dialog eats mouse events which results in the close button
+  // getting stuck in the hover state. Reset the window controls to
+  // prevent this.
+  BrowserView::GetBrowserViewForBrowser(browser)
+      ->GetWidget()
+      ->non_client_view()
+      ->ResetWindowControls();
```

Comment:
We don't need handling for this fallback dialog as fallback dialog is not launched from window control buttons.
It's needed for window closing dialog as it's launched when click window close button.